scc/light_client/provider/rpc_net_test.go

```diff
@@ -0,0 +1,147 @@
+package provider
+
+import (
+	"fmt"
+	"math"
+	"math/big"
+	"testing"
+
+	"github.com/0xsoniclabs/sonic/tests"
+	"github.com/0xsoniclabs/sonic/tests/contracts/counter_event_emitter"
+	"github.com/Fantom-foundation/lachesis-base/inter/idx"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ethereum/go-ethereum/ethclient"
+	"github.com/ethereum/go-ethereum/rpc"
+	"github.com/stretchr/testify/require"
+)
+
+////////////////////////////////////////
+// integration net tests
+////////////////////////////////////////
+
+func TestRpcProvider_GetCommitteeCertificates_CanRetrieveCertificates(t *testing.T) {
+	require := require.New(t)
+
+	// start network
+	net, client := startNetAndGetClient(t)
+
+	// make providers
+	providerFromClient := NewRpcProviderFromClient(client.Client())
+	t.Cleanup(providerFromClient.Close)
+	url := fmt.Sprintf("http://localhost:%d", net.GetPort())
+	providerFromURL, err := NewRpcProviderFromURL(url)
+	require.NoError(err)
+	t.Cleanup(providerFromURL.Close)
+
+	chainId := getChainId(t, client.Client())
+
+	for _, provider := range []*RpcProvider{providerFromClient, providerFromURL} {
+
+		// get certificates
+		certs, err := provider.GetCommitteeCertificates(0, math.MaxUint64)
+		require.NoError(err)
+
+		require.NotZero(len(certs))
+		for _, cert := range certs {
+			require.Equal(chainId.Uint64(), cert.Subject().ChainId)
+		}
+	}
+}
+
+func TestRpcProvider_GetBlockCertificates_CanRetrieveCertificates(t *testing.T) {
+	require := require.New(t)
+
+	// start network
+	net, client := startNetAndGetClient(t)
+
+	// Produce a few blocks on the network. We use the counter contract since
+	// it is also producing events.
+	const numBlocks = 10
+	counter, _, err := tests.DeployContract(net, counter_event_emitter.DeployCounterEventEmitter)
+	require.NoError(err)
+	for range numBlocks {
+		_, err := net.Apply(counter.Increment)
+		require.NoError(err, "failed to increment counter")
+	}
```

Comment:
This seems to be irrelevant for this test. In any test chain there should be at least 2 blocks from the genesis.

Reply:
https://github.com/0xsoniclabs/sonic/pull/91/commits/176a41305ce922ce0aa51d133d879bd9a2347161